agent/config/parse_windows.go

```diff
@@ -71,3 +89,24 @@ func isDomainJoined() (bool, error) {
 
 	return status == syscall.NetSetupDomainName, nil
 }
+
+// isWindows2016 is used to check if container instance is versioned Windows 2016
+// Reference: https://godoc.org/golang.org/x/sys/windows/registry
+func isWindows2016() bool {
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		seelog.Errorf("Unable to open Windows registry key to determine Windows version: %v", err)
+	}
+	defer key.Close()
+
+	version, _, err := key.GetStringValue("ProductName")
+	if err != nil {
+		seelog.Errorf("Unable to read current version from Windows registry: %v", err)
```

Comment:
Is it worth checking `version` if there is an error here? Should probably just return `false`

Reply:
synced offline.